src/targeting.js

```diff
@@ -464,6 +464,14 @@ export function newTargeting(auctionManager) {
       if (typeof filterFunction === 'function') {
         bidsReceived = bidsReceived.filter(bid => latestAuctionForAdUnit[bid.adUnitCode] === bid.auctionId || !!filterFunction(bid))
       }
+
+      bidsReceived = bidsReceived
+        .map(bid => {
+          if (latestAuctionForAdUnit[bid.adUnitCode] !== bid.auctionId) {
+            bid.latestAuctionForAdUnit = latestAuctionForAdUnit[bid.adUnitCode];
```

Comment:
- IMO this should be set on all bids in the pool, not just those from a previous auction.
- I don't like the name, "forAdUnit" is redundant since the object is a bid and already tied to a particular ad unit. Maybe `latestTargetedAuctionId`? Asking also @patmmccann for opinions. I'd prefer to make targeting explicit since this only runs when the publisher uses prebid's targeting logic, which in theory they could do without.

Reply:
On naming , I agree with a shorter name yes